owncloud/owncloud.py

```diff
@@ -804,6 +1022,105 @@ def delete_attribute(self, app, key):
+	#  <data><apps><element>files</element><element>activity</element> ...
+        for el in tree.findall('data/apps/element'):
+	  ena_apps[el.text] = False
+
+        res = self.__make_ocs_request('GET', self.OCS_SERVICE_CLOUD, 'apps?filter=enabled')
+        if res.status_code != 200:
+	    raise ResponseError(res)
+        tree = ET.fromstring(res.text)
+        self.__check_ocs_status(tree)
+        for el in tree.findall('data/apps/element'):
+	  ena_apps[el.text] = True
+
+        return ena_apps	
+
+
+    def enable_app_ajax(self, adminuser, adminpass, appname):
+        """Enable an app through the ajax calls for the web browser.
+
+	   appname in (provisioning_api admin_dependencies_chk activity files_trashbin enterprise_key
+	     admin_migrate files_locking firewall firstrunwizard templateeditor files_sharing files_texteditor
+	     files_versions files_antivirus files_encryption external files_external user_external 
+	     files_sharing_log files_ldap_home user_ldap admin_audit admin_dependencies_chk sharepoint 
+	     user_shibboleth user_webdavauth windows_network_drive)
+
+	   POST http://192.168.122.19/owncloud/index.php/settings/ajax/enableapp.php
+
+	   Host: 192.168.122.19
+	   User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:31.0) Gecko/20100101 Firefox/31.0
+	   Accept: application/json, text/javascript, */*; q=0.01
+	   Accept-Language: en-us,de;q=0.7,en;q=0.3
+	   Accept-Encoding: gzip, deflate
+	   Content-Type: application/x-www-form-urlencoded; charset=UTF-8
+	   requesttoken: 6c6ee0722d72b8b56270
+	   X-Requested-With: XMLHttpRequest
+	   Referer: http://192.168.122.19/owncloud/index.php/settings/apps?installed
+	   Content-Length: 28
+	   Cookie: ocbd0b2139bb=avgk4032thgaknqdmu22emulh4e87821l9elhff9ha1dp5p3cg20; ocbd0b2139bb=rl8r4bn9spvdlur63vsdlmoq8juel5a4fo0snaoaqkkiglt5m4j0
+	   Connection: keep-alive
+	   Pragma: no-cache
+	   Cache-Control: no-cache
+
+	   appid=admin_dependencies_chk
+	"""
+        res = self.__make_single_request(
+                'POST',
+                'index.php/settings/ajax/enableapp.php',
+		data = { 'appid': appname },
+		auth = { 'username': adminuser, 'password': adminpass }
+	        )
+        if res.status_code == 200:
+	  return res.text
+        raise ResponseError(res)
+
+
+    def disable_app_ajax(self, adminuser, adminpass, appname):
+        """Disable an app through the ajax calls for the web browser.
+	"""
+        res = self.__make_single_request(
+                'POST',
+                'index.php/settings/ajax/disableapp.php',
+		data = { 'appid': appname },
+		auth = { 'username': adminuser, 'password': adminpass }
+	        )
+        if res.status_code == 200:
+	  return res.text
+        raise ResponseError(res)
+
+
+    def enable_api(self, appname):
+        """Enable an app through provisioning_api
+	"""
+        res = self.__make_ocs_request('POST', self.OCS_SERVICE_CLOUD, 'apps/'+appname)
+        if res.status_code == 200:
+	  return res.text
```

Comment:
Not sure what this returns, need to check and document, or simply use `return` with nothing